code/game/turfs/walls/walls.dm

```diff
@@ -61,6 +63,11 @@
 	update_connections(FALSE)
 	update_icon()
 
+/turf/closed/wall/Destroy(force)
+	if(leaned_on_by)
+		stop_wall_lean()
```

Comment:
This is Destroy(), clear the var after so it still works independently of logic or runtimes. In theory you're not even supposed to have side effects such as this in Destroy :S